### PR TITLE
Updated to Spring Boot 2.1.4.RELEASE

### LINKED ISSUE #8: Consider adding Spring Boot 2 support
Relates to https://github.com/spring-projects/spring-boot/issues/13510

We are reviewing our [community starters list](https://github.com/spring-projects/spring-boot/blob/7683d346e31bbe0313a03ce487ee59454665a143/spring-boot-project/spring-boot-starters/README.adoc) and a quick look here suggests that this starter doesn’t yet support Spring Boot 2.

Can you please let us know if/when you plan to support Spring Boot 2?

Thanks!

### PR DESCRIPTION
**Pull request by santam85**
I updated the starter to be compatible with Spring Boot 2.1.4.RELEASE
Fixes #8

### CODE REVIEW

**Comment by santam85** [comment]
@hzpz can you review this at your earliest convenience?

**Comment by hzpz** [comment]
Thank you! I'll have a look.

File: pom.xml [hunk 1]
```diff
@@ -4,15 +4,15 @@
 
     <groupId>io.github.hzpz.spring.boot</groupId>
     <artifactId>mongeez-spring-boot-starter</artifactId>
-    <version>1.4.0-SNAPSHOT</version>
+    <version>2.1.5-SNAPSHOT</version>
```
Comment (hzpz) [inline pom.xml]: I don't see why the version of the starter should change. I _might_ make it 2.0.0, but the PR should leave it as is.

File: src/main/java/io/github/hzpz/spring/boot/autoconfigure/mongeez/MongoDbFactoryDependsOnPostProcessor.java [hunk 2]
```diff
@@ -57,15 +47,24 @@ private static BeanDefinition getBeanDefinition(String beanName,
...
             if (parentBeanFactory instanceof ConfigurableListableBeanFactory) {
                 return getBeanDefinition(beanName,
-                        (ConfigurableListableBeanFactory) parentBeanFactory);
+                    (ConfigurableListableBeanFactory) parentBeanFactory);
             }
             throw ex;
         }
     }
 
+    public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory) {
```
Comment (hzpz) [inline src/main/java/io/github/hzpz/spring/boot/autoconfigure/mongeez/MongoDbFactoryDependsOnPostProcessor.java]: All changes in this class seem to be purely cosmetical. I don't see a reason for that, please revert those changes.

File: src/test/java/io/github/hzpz/spring/boot/autoconfigure/mongeez/MongeezAutoConfigurationTests.java [hunk 3]
```diff
@@ -63,70 +64,78 @@ public void shouldDoNothingIfDisabled() {
...
     public void shouldUseDatabaseFromMongoProperties() {
         String database = "foo";
-        EnvironmentTestUtils.addEnvironment(this.context, "spring.data.mongodb.database:" + database);
+        TestPropertyValues.of("spring.data.mongodb.database:" + database).applyTo(this.context);
         registerAndRefresh(DoNotExecuteMongeezPostProcessor.class,
-                MongoAutoConfiguration.class, MongeezAutoConfiguration.class);
+            MongoAutoConfiguration.class, MongeezAutoConfiguration.class);
         Mongeez mongeez = this.context.getBean(Mongeez.class);
         Object mongeezDatabase = ReflectionTestUtils.getField(mongeez, "dbName");
+        assertNotNull(mongeezDatabase);
```
Comment (hzpz) [inline src/test/java/io/github/hzpz/spring/boot/autoconfigure/mongeez/MongeezAutoConfigurationTests.java]: Why did you add this assert? If that fails then something is fundamentally wrong with this test. Therefore it should not "fail" (in JUnit terms) but error out. Which would happen anyway in the next line (caused by a `NullPointerException`).

File: src/test/java/io/github/hzpz/spring/boot/autoconfigure/mongeez/MongeezAutoConfigurationTests.java [hunk 4]
```diff
@@ -63,70 +64,78 @@ public void shouldDoNothingIfDisabled() {
     @Test
     public void shouldUseDatabaseFromMongoProperties() {
         String database = "foo";
-        EnvironmentTestUtils.addEnvironment(this.context, "spring.data.mongodb.database:" + database);
+        TestPropertyValues.of("spring.data.mongodb.database:" + database).applyTo(this.context);
         registerAndRefresh(DoNotExecuteMongeezPostProcessor.class,
-                MongoAutoConfiguration.class, MongeezAutoConfiguration.class);
+            MongoAutoConfiguration.class, MongeezAutoConfiguration.class);
```
Comment (hzpz) [inline src/test/java/io/github/hzpz/spring/boot/autoconfigure/mongeez/MongeezAutoConfigurationTests.java]: You seem to be using a different formatter than me. Please make sure, that you don't submit any formatting changes like this.

File: src/test/java/io/github/hzpz/spring/boot/autoconfigure/mongeez/MongeezAutoConfigurationTests.java [hunk 5]
```diff
@@ -63,70 +64,78 @@ public void shouldDoNothingIfDisabled() {
...
-        EnvironmentTestUtils.addEnvironment(this.context, "mongeez.database:" + mongeezOverrideDatabase);
+        TestPropertyValues.of("spring.data.mongodb.database:" + mongoDatabase)
+            .and("mongeez.database:" + mongeezOverrideDatabase)
+            .applyTo(this.context);
         registerAndRefresh(DoNotExecuteMongeezPostProcessor.class,
-                MongoAutoConfiguration.class, MongeezAutoConfiguration.class);
+            MongoAutoConfiguration.class, MongeezAutoConfiguration.class);
         Mongeez mongeez = this.context.getBean(Mongeez.class);
         Object mongeezActualDatabase = ReflectionTestUtils.getField(mongeez, "dbName");
+        assertNotNull(mongeezActualDatabase);
```
Comment (hzpz) [inline src/test/java/io/github/hzpz/spring/boot/autoconfigure/mongeez/MongeezAutoConfigurationTests.java]: See above.

File: src/test/java/io/github/hzpz/spring/boot/autoconfigure/mongeez/MongeezAutoConfigurationTests.java [hunk 6]
```diff
@@ -63,70 +64,78 @@ public void shouldDoNothingIfDisabled() {
...
+            .and("mongeez.password:pass")
+            .applyTo(this.context);
         registerAndRefresh(DoNotExecuteMongeezPostProcessor.class,
-                MongoAutoConfiguration.class, MongeezAutoConfiguration.class);
+            MongoAutoConfiguration.class, MongeezAutoConfiguration.class);
         Mongeez mongeez = this.context.getBean(Mongeez.class);
-        String authDb = ((MongoAuth) ReflectionTestUtils.getField(mongeez, "auth")).getAuthDb();
-        assertThat(authDb, equalTo(database));
+        MongoAuth auth = (MongoAuth) ReflectionTestUtils.getField(mongeez, "auth");
+        assertNotNull(auth);
```
Comment (hzpz) [inline src/test/java/io/github/hzpz/spring/boot/autoconfigure/mongeez/MongeezAutoConfigurationTests.java]: See above.

File: src/test/java/io/github/hzpz/spring/boot/autoconfigure/mongeez/MongeezAutoConfigurationTests.java [hunk 7]
```diff
@@ -63,70 +64,78 @@ public void shouldDoNothingIfDisabled() {
...
+            .and("mongeez.password:pass")
+            .applyTo(this.context);
         registerAndRefresh(DoNotExecuteMongeezPostProcessor.class,
-                MongoAutoConfiguration.class, MongeezAutoConfiguration.class);
+            MongoAutoConfiguration.class, MongeezAutoConfiguration.class);
         Mongeez mongeez = this.context.getBean(Mongeez.class);
-        String authDb = ((MongoAuth) ReflectionTestUtils.getField(mongeez, "auth")).getAuthDb();
-        assertThat(authDb, equalTo(mongeezOverrideDatabase));
+        MongoAuth auth = (MongoAuth) ReflectionTestUtils.getField(mongeez, "auth");
+        assertNotNull(auth);
```
Comment (hzpz) [inline src/test/java/io/github/hzpz/spring/boot/autoconfigure/mongeez/MongeezAutoConfigurationTests.java]: See above.

**Review by hzpz — CHANGES_REQUESTED** [review]

**Comment by santam85** [comment]
@hzpz changes addressed.

**Review by hzpz — APPROVED** [review]

**Comment by hzpz** [comment]
Thanks again. I'll create and publish a new release in the next few days.

**Comment by santam85** [comment]
Hello @hzpz, I noticed the release for this is still pending, anything I can do to to help expedite?

**Comment by hzpz** [comment]
Hey @santam85, sorry for the delay, see #12

**Comment by santam85** [comment]
Thanks!

**Pull request merged**